tests/unit/test1_unit_tests/src/main.c

```diff
@@ -143,12 +143,312 @@ void test_size64(void)
+
+	zassert_equal(198, output.len, NULL);
+	zassert_mem_equal(output.value,
+		"HelloWorld1HelloWorld2HelloWorld3HelloWorld4HelloWorld5HelloWorld6" \
+		"HelloWorld7HelloWorld8HelloWorld9HelloWorldAHelloWorldBHelloWorldC" \
+		"HelloWorldDHelloWorldEHelloWorldFHelloWorldGHelloWorldHHelloWorldI",
+		198, NULL);
+
+	// Start negative tests
+
+	zassert_false(zcbor_validate_string_fragments(output_frags, 17), NULL); // Last fragment missing
+
+	output_frags[2].total_len--;
+	zassert_false(zcbor_validate_string_fragments(output_frags, 18), NULL); // Mismatch total_len
+	output_frags[2].total_len += 2;
+	zassert_false(zcbor_validate_string_fragments(output_frags, 18), NULL); // Mismatch total_len
+	output_frags[2].total_len--;
+
+	memcpy(&output_frags_backup, &output_frags[16], sizeof(output_frags[0]));
+	memcpy(&output_frags[16], &output_frags[17], sizeof(output_frags[0])); // Move last fragment
+
+	zassert_false(zcbor_validate_string_fragments(output_frags, 17), NULL); // Second-to-last fragment missing
+	memcpy(&output_frags[16], &output_frags_backup, sizeof(output_frags[0])); // Restore
+
+	output_frags[6].offset++;
+	output_frags[7].offset--;
+	zassert_false(zcbor_validate_string_fragments(output_frags, 18), NULL); // Slight error in offset vs. fragment lengths.
+	output_frags[6].offset--;
+	output_frags[7].offset++;
+
+	for (int i = 0; i < 18; i++) {
+		output_frags[i].total_len -= output_frags[17].fragment.len;
+	}
+
+	zassert_true(zcbor_validate_string_fragments(output_frags, 17), NULL); // Should work with 17 fragments now.
+	zassert_false(zcbor_validate_string_fragments(output_frags, 18), NULL); // Last fragment will extend past total_len.
+
+	for (int i = 0; i < 18; i++) {
+		output_frags[i].total_len += (output_frags[17].fragment.len - 1);
+	}
+	zassert_false(zcbor_validate_string_fragments(output_frags, 18), NULL); // Last fragment will extend past total_len by a single byte.
+	output_frags[17].fragment.len--;
+	zassert_true(zcbor_validate_string_fragments(output_frags, 18), NULL); // Should work with shorter last fragment
+	output_frags[17].fragment.len++; // Restore
+	for (int i = 0; i < 18; i++) {
+		output_frags[i].total_len += 1; // Restore
+	}
+
+	zassert_true(zcbor_validate_string_fragments(output_frags, 18), NULL); // Check that all errors were restored correctly.
+}
+
+
+/** This test creates the following structure, wrapped in a BSTR:
+ *
+ *  (
+ *      42,
+ *      "Hello World",
+ *      [
+ *          true,
+ *          nil,
+ *      ]
+ *  )
+ *
+ *  This structures is then split in three fragments (output_frags) like so:
+ *  1st fragment (8 bytes):
+ *  (
+ *      42,
+ *      "Hell
+ *
+ *  2nd fragment (8 bytes):
+ *           o World",
+ *      [
+ *
+ *  3rd fragment (3 bytes or 2 bytes if ZCBOR_CANONICAL is defined):
+ *          true,
+ *          nil,
+ *      ]
+ *  )
+ *
+ *  This means that the TSTR has to be handled using fragments (tstr_frags)
+ *  as well.
+ */
+void test_bstr_cbor_fragments(void)
```

Comment:
This is a bit unrelated but I'm trying to see how this would help me with receiving a FOTA update - [LwM2M CBOR content format support](https://github.com/nrfconnect/sdk-zephyr/blob/3ed1f805d2c04a930eeb04907d3a580d37ea816f/subsys/net/lib/lwm2m/lwm2m_rw_cbor.c#L390) - but I don't. I wouldn't personally need reassembly, just a way to take a peek at the CBOR header without a range check. FOTA is done with CoAP blockwise transfer where only the first message has the CBOR header and the rest of the blocks have opaque data.

Reply:
Yeah, I don't assume you need all of these, probably just `zcbor_bstr_start_decode_fragment()` which hopefully does what you want, and `zcbor_update_state()` which you need to keep everything up to date when you add new blocks, and then `zcbor_bstr_end_decode()` at the end. At least that's the intended way.

Anything else is just if it makes your life easier.